examples/reflection/custom_attributes.rs

```diff
@@ -0,0 +1,90 @@
+//! Demonstrates how to register and access custom attributes on reflected types.
+
+use bevy::reflect::{Reflect, TypeInfo, Typed};
+use std::any::TypeId;
+use std::ops::RangeInclusive;
+
+fn main() {
+    // Bevy supports statically registering custom attribute data on reflected types,
+    // which can then be accessed at runtime via the type's `TypeInfo`.
+    // Attributes are registered using the `#[reflect(@...)]` syntax,
+    // where the `...` is any expression that resolves to a value which implements `Reflect`.
+    // Note that these attributes are stored based on their type:
+    // if two attributes have the same type, the second one will overwrite the first.
+
+    // Here is an example of registering custom attributes on a type:
+    #[derive(Reflect)]
+    struct Slider {
+        #[reflect(@RangeInclusive::<f32>::new(0.0, 1.0))]
```

Comment:
Nice F12 / docs on hover works here, as theorized!
Autocomplete doesn't currently work, but we might actually be able to fix that with some macro changes (still not certain if RA will support this scenario for struct attributes). Not necessary for a V1 though.